lua/navigator/hierarchy.lua

```diff
@@ -253,22 +253,34 @@ end
 
 local request = vim.lsp.buf_request
 
+local function pick_hierarchy_lsp_client(bufnr, method)
+  bufnr = bufnr or 0
+  method = method or vim.lsp.protocol.Methods.textDocument_prepareCallHierarchy
+  local file = vim.api.nvim_buf_get_name(bufnr)
+
+  for _, client in ipairs(vim.lsp.get_clients({ bufnr = bufnr, method = method })) do
+    if client.server_capabilities.callHierarchyProvider then
+      local root = client.config and client.config.root_dir
+      if not root or file:find(root, 1, true) == 1 then
```

Comment:
should it be 
```lua
if root and file:find(root, 1, true) == 1
```
Is there an example of lsp returns nil value for root_dir?

Reply:
My thoughts here were to not enforce `root` to cover edge cases where a bad configuration or architectural decisions of some LSP's could lead to a nil `root_dir` since the protocol allows it. Personally, I've no encounter these cases since all the LSP's that I use always have a defined `root_dir` but I'll let you decide here, if you think it's better to enforce `root` I can change it, let me know.